src/common/debug-requests.ts

```diff
@@ -0,0 +1,39 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+// inspired by https://github.com/eclipse-theia/theia/blob/master/packages/debug/src/browser/debug-session-connection.ts
+
+import type { DebugProtocol } from '@vscode/debugprotocol';
+import type { DebugSession } from 'vscode';
+
+export interface DebugRequestTypes {
+    'evaluate': [DebugProtocol.EvaluateArguments, DebugProtocol.EvaluateResponse['body']]
+    'readMemory': [DebugProtocol.ReadMemoryArguments, DebugProtocol.ReadMemoryResponse['body']]
+    'writeMemory': [DebugProtocol.WriteMemoryArguments, DebugProtocol.WriteMemoryResponse['body']]
+}
```

Comment:
Absolutely true! I wonder if the VS Code people had a discussion about that because as soon as you actually start using it, you really see how accessing the `body` of the responses makes for really long and really weird types ;-)

So my thought process in the split between the `debug-requests.ts` and `messaging.ts` is that the messaging is really the protocol between the extension's main side and the extension webview whereas the debug protocol messages are between the extension's main side and the debug adapter.

Therefore, bleeding any types from the debug protocol into the webview does not make much sense as it cannot directly access the debug adapter. Following that, it seemed logical to have a messaging layer in-between where we define the arguments and results of the main-webview communication. Sure, some of those types may be a 1:1 alias for a debug protocol message but we may also want to extend them at some point (like I do with the `WriteMemoryArguments`), make parts required or optional (with a lot of the `Partial` going around) or re-define them. So in a way, having all those `DebugProtocol.ReadMemoryArguments` in the webview and the options is really more a "coincidence" because it obviously represents memory reading well but and also avoids giving some of the usages a more specific semantic imho.

Based on that, I'm absolutely not opposed to using the `DebugProtocol` interfaces on the extension's main side within all the debug-related classes (which is why I also left them in the `memory-provider`). Of course I wished for their names to be shorter, so we could alias them as in the `debug-requests.ts` as you suggested but for me personally, that would not replace the type definitions that we do in `messaging.ts`.

Long story short: I think we should distinguish between the types used in the main-debug communication and the ones in the main-webview communication. However, maybe I just overthought the whole thing so I don't plan to die on this hill ;-) I'd be very happy to get your thoughts on that though.